objects/obj_controller/Create_0.gml

```diff
@@ -1648,3 +1648,422 @@ if (vih>=5){
 remov=string_length(string(temp[65])+string(temp[66])+string(temp[67])+string(temp[68])+string(temp[69]))+1;
 
 action_set_alarm(2, 0);
+
+#region serialization
+
+serialize = function() {
+
+    // general data and inits
+
+    var _ctrl_data = {
+        play_time,
+        cheatyface,
+        x,
+        y,
+        alll,
+        zoom: {
+            was_zoomed,
+            zoomed
+        },
+        chaos_rating,
+        fleet_type,
+        homeworld_rule,
+        star_names,
+        craftworld,
+        turn,
+        last_event,
+        last_mission,
+        last_world_inspection,
+        last_fleet_inspection,
+        chaos_turn,
+        chaos_fleets,
+        tau_fleets,
+        tau_stars,
+        tau_messenger,
+        fleet_all,
+        unload,
+        diplomacy,
+        trading,
+        audience,
+        force_goodbye,
+        combat,
+        new_vehicles,
+        hurssy,
+        hurssy_time,
+        artifacts,
+        popup_master_crafted,
+        select_wounded,
+        terra_direction,
+        stc: {
```

Comment:
as you can see, I reorganized variables a little, I wanted to see how a more data-driven approach would look like, and I like it so far